### PR TITLE
Add config defined threshold kicks to Bouncer

### PR DESCRIPTION
**Pull request by moisterrific**
along with various other minor refinements

### CODE REVIEW

**Comment by moisterrific** [comment]
oh 
image

File: TShockAPI/Bouncer.cs [hunk 1]
```diff
@@ -494,19 +494,41 @@ internal void OnTileEdit(object sender, GetDataHandlers.TileEditEventArgs args)
...
+						TShock.Log.ConsoleDebug("If this player wasn't hacking, please report the tile kill threshold they were disabled for to TShock so we can improve this!");
+						args.Handled = true;
+						return;
+					}
+					else
+					{
+						TShock.Log.ConsoleDebug("Bouncer / OnTileEdit rejected from tile kill threshold from {0}, (value: {1})", args.Player.Name, args.Player.TileKillThreshold);
+						TShock.Log.ConsoleDebug("If this player wasn't hacking, please report the tile kill threshold they were disabled for to TShock so we can improve this!");
+						args.Player.Disable("Reached TileKill threshold.", DisableFlags.WriteToLogAndConsole);
+					}
```
Comment (QuiCM) [inline TShockAPI/Bouncer.cs:505]: You could condense this if/else block:
```csharp
					if (TShock.Config.KickOnTileKillThresholdBroken)
					{
						args.Player.Kick(string.Format("Tile kill threshold exceeded {0}.", TShock.Config.TileKillThreshold));
					}
					else
					{
						args.Player.Disable("Reached TileKill threshold.", DisableFlags.WriteToLogAndConsole);
					        args.Player.SendTileSquare(tileX, tileY, 4);
					}

					TShock.Log.ConsoleDebug("Bouncer / OnTileEdit rejected from tile kill threshold from {0}, (value: {1})", args.Player.Name, args.Player.TileKillThreshold);
					TShock.Log.ConsoleDebug("If this player wasn't hacking, please report the tile kill threshold they were disabled for to TShock so we can improve this!");
					args.Handled = true;
```

File: TShockAPI/Bouncer.cs [hunk 2]
```diff
@@ -494,19 +494,41 @@ internal void OnTileEdit(object sender, GetDataHandlers.TileEditEventArgs args)
...
 					args.Handled = true;
 					return;
 				}
 
 				if (args.Player.TilePlaceThreshold >= TShock.Config.TilePlaceThreshold)
 				{
-					TShock.Log.ConsoleDebug("Bouncer / OnTileEdit rejected from tile place threshold from {0}, (value: {1})", args.Player.Name, args.Player.TilePlaceThreshold);
-					TShock.Log.ConsoleDebug("If this player wasn't hacking, please report the damage value they were disabled for to TShock so we can improve this!");
-					args.Player.Disable("Reached TilePlace threshold.", DisableFlags.WriteToLogAndConsole);
+					if (TShock.Config.KickOnTilePlaceThresholdBroken)
```
Comment (QuiCM) [inline TShockAPI/Bouncer.cs:515]: See above

File: TShockAPI/Bouncer.cs [hunk 3]
```diff
@@ -761,8 +783,20 @@ internal void OnNewProjectile(object sender, GetDataHandlers.NewProjectileEventA
 
 			if (args.Player.ProjectileThreshold >= TShock.Config.ProjectileThreshold)
 			{
-				args.Player.Disable("Reached projectile update threshold.", DisableFlags.WriteToLogAndConsole);
-				TShock.Log.ConsoleDebug("Bouncer / OnNewProjectile rejected from proj update threshold from {0} {1}/{2}", args.Player.Name, args.Player.ProjectileThreshold, TShock.Config.ProjectileThreshold);
+				if (TShock.Config.KickOnProjectileThresholdBroken)
```
Comment (QuiCM) [inline TShockAPI/Bouncer.cs:786]: See above

File: TShockAPI/Bouncer.cs [hunk 4]
```diff
@@ -1114,8 +1148,20 @@ internal void OnLiquidSet(object sender, GetDataHandlers.LiquidSetEventArgs args
 
 			if (args.Player.TileLiquidThreshold >= TShock.Config.TileLiquidThreshold)
 			{
-				TShock.Log.ConsoleDebug("Bouncer / OnLiquidSet rejected from liquid threshold from {0} {1}/{2}", args.Player.Name, args.Player.TileLiquidThreshold, TShock.Config.TileLiquidThreshold);
-				args.Player.Disable("Reached TileLiquid threshold.", DisableFlags.WriteToLogAndConsole);
+				if (TShock.Config.KickOnTileLiquidThresholdBroken)
```
Comment (QuiCM) [inline TShockAPI/Bouncer.cs:1141]: See above

File: TShockAPI/Bouncer.cs [hunk 5]
```diff
@@ -1442,23 +1488,35 @@ internal void OnHealOtherPlayer(object sender, GetDataHandlers.HealOtherPlayerEv
...
 				args.Handled = true;
 				return;
 			}
 
-			if (args.Player.HealOtherThreshold > TShock.Config.HealOtherThreshold)
+			if (args.Player.HealOtherThreshold >= TShock.Config.HealOtherThreshold)
 			{
-				TShock.Log.ConsoleDebug("Bouncer / OnUpdateNPCHome rejected heal other threshold from {0} {1}/{2}", args.Player.Name, args.Player.HealOtherThreshold, TShock.Config.HealOtherThreshold);
-				args.Player.Disable("Reached HealOtherPlayer threshold.", DisableFlags.WriteToLogAndConsole);
+				if (TShock.Config.KickOnHealOtherThresholdBroken)
```
Comment (QuiCM) [inline TShockAPI/Bouncer.cs:1486]: See above :)

File: TShockAPI/Bouncer.cs (hunk 1 above)
Comment (moisterrific) [reply TShockAPI/Bouncer.cs:505]: Oh yea that looks much cleaner and makes more sense. I was just following how `OnNPCStrike` did it lol

image

Comment (moisterrific) [reply TShockAPI/Bouncer.cs:505]: I'll update that while I'm at it

**Review by QuiCM — APPROVED** [review]

**Pull request merged**